### PR TITLE
added the auth token for increasing api-rate-limit

### LINKED ISSUE #7492: Fix needed for the rate limit issue in the GitHub API endpoint 'users/{userName}' within the 'getLocationByUsername.js' file
### Description

During my code analysis, I discovered that we are utilizing the GitHub API "**_https://api.github.com/users/{userName}_**" and employing the Octokit utility provided by GitHub for making the API calls.

However, I noticed that we are not supplying an authentication token when creating the Octokit object. This is crucial because the API has a rate limiter implemented, which restricts the number of user reads to 60 per hour. Given our requirements, this limit is insufficient.

Here is a screenshot of the relevant code for reference from "getLocationByUsername.js".

image

See the error while loading the test data. 

image

**Proposed Solution:**
-------------------------------------------------------------------------------------------------------------
According to the GitHub rate limiter guidelines, if we include an authentication token when making requests to this API, we will be allowed to make up to 5000 requests per hour. This limit appears to be acceptable for the current situation. However, as the number of users grows, we may need to reconsider this aspect in the future. 

To address this issue, let's fix it by adding an authentication token as shown below:

image

Have a look into detailed documentation for rate limit from GitHub https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limiting

### Screenshots

_No response_

### Additional information

Please consult the following document, [Authentication Doc](https://github.com/octokit/auth-app.js/#createappauthoptions-or-new-octokit-auth-), for detailed instructions on the necessary steps to implement token authentication. This guide outlines the specific procedures required to obtain and utilize the token for reading purposes.

Screenshot for generating the token-

image

More reference:  Follow this youtube video for step by step process. 

[How to use GitHub REST API](https://www.youtube.com/watch?v=OvfLavRD1Os)

Happy Coding!

### PR DESCRIPTION
**Pull request by Ankur1493**
## Fixes Issue

Closes #7492 

## Changes proposed

As said in the proposed solution I have added the auth token in the getLocationByUsername and have tested by adding a personal access token and making this call "[https://api.github.com/users/{userName}" 
I have to add a personal access token to .env file so I think it requires a docs change sorry don't know that much about it 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [x] My change requires changes to the documentation.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Screenshots

Screenshot 2023-06-11 09 49 30
Screenshot 2023-06-11 18 30 36
Screenshot 2023-06-11 18 30 55
Screenshot 2023-06-11 18 34 37

## Note to reviewers

This was my first contribution to Linkfree Thanks for assigning this to me. I have never got assigned an issue this fast. It's great working with the Eddiehub community

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7521"></a>

### CODE REVIEW

File: services/github/getLocationByUsername.js [hunk 1]
```diff
@@ -9,7 +9,9 @@ export default async function getLocationByUsername(username) {
     lat: 0,
     lon: 0,
   };
-  const octokit = new Octokit({});
+  const octokit = new Octokit({
+    auth: "process.env.GITHUB_API_TOKEN"
```
Comment (raviraj-github) [inline services/github/getLocationByUsername.js:13]: Please make the necessary updates to the .env, .env.example, and .env.docker files by adding the environment variable GITHUB_API_TOKEN. The default value for this variable should be an empty string.

Comment (eddiejaoude) [reply services/github/getLocationByUsername.js:13]: In addition to Ravi's comment

- `process.env.GITHUB_API_TOKEN` should not be quoted
- we should not be using `process`, please check the `config/schemas/serverSchema.js`

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for your contribution 👍 

We have left some small inline comments, and also please update the environment variable docs

**Comment by eddiejaoude** [comment]
@Ankur1493 did you mean to close this PR?

**Comment by Ankur1493** [comment]
yes I have created a new one all fresh this one had many commits rebased and was rushed because of excitement

**Pull request closed without merging**